### PR TITLE
[wgsl] Minor formatting and language changes.

### PR DESCRIPTION
**Pull request by dj2**
This PR fixes a few spelling issues, adds more linking into various
parts of the spec and updates/modifies some of the language and
punctuation in various parts of the spec.

### CODE REVIEW

File: wgsl/index.bs [hunk 1]
```diff
@@ -1023,13 +1023,13 @@ Each distinct type parameterization for a type rule is called an <dfn noexport>o
 For example, [[#arithmetic-expr|unary negation]] (an expression of the form `-`|e|)
 has eight overloads, because its type rules are parameterized by a type |T| that can be any of:
 * [=i32=]
-* vec2&lt;i32&gt;
-* vec3&lt;i32&gt;
-* vec4&lt;i32&gt;
+* [=vector|vec2&lt;i32&gt;=]
```
Comment (alan-baker) [inline wgsl/index.bs:1026]: One link for the vectors is probably enough and will help prevent readers from thinking that these links might be to different parts of the spec.

**Review by alan-baker — CHANGES_REQUESTED** [review]

File: wgsl/index.bs [hunk 2]
```diff
@@ -6230,7 +6230,8 @@ Additionally, certain statements may only be used in very specific contexts:
 ### Rules ### {#behaviors-rules}
 
 Some statements affecting control-flow are only valid in some contexts.
-For example, `fallthrough` is invalid outside of a switch, and `continue` is invalid outside of a loop.
+For example, [=statement/fallthrough=] is invalid outside of a [=statement/switch=],
+and [=statement/continue=] is invalid outside of a [=statement/loop=], [=statement/for=] or [=statement/while=].
```
[the remark this answers is not in the document]
Comment (dj2) [reply wgsl/index.bs]: Done, although I prefer the Random Comma ...

File: wgsl/index.bs [hunk 3]
```diff
@@ -308,8 +308,8 @@ The events are:
         and invokes the [=shader stage=] [=entry point=] function.
 4. <dfn noexport>Shader execution end</dfn>
     * This occurs when all work in the shader completes:
-        * all its [=invocations=] terminate
-        * and all accesses to [=resources=] complete
+        * all its [=invocations=] terminate.
+        * all accesses to [=resources=] complete.
         * outputs, if any, are passed to downstream pipeline stages.
```
[the remark this answers is not in the document]
Comment (alan-baker) [reply wgsl/index.bs:313]: It might just be my confusion, but I think that speaks to making it explicit.

[the remark this answers is not in the document]
Comment (dj2) [reply wgsl/index.bs:313]: Fair enough.

File: wgsl/index.bs (hunk 1 above)
Comment (dj2) [reply wgsl/index.bs:1026]: done.

File: wgsl/index.bs [hunk 4]
```diff
@@ -2102,57 +2102,57 @@ following table:
...
+  <tr><td>[=vector|vec=]4&lt;|T|&gt;
       <td>16
       <td>16
-  <tr><td>mat|N|x|M| (col-major)<br>
+  <tr><td>[=matrix|mat=]|N|x|M| (col-major)<br>
       <p class="small">(General form)</p>
       <td>[=AlignOf=](vec|M|)
       <td>[=SizeOf=](array&lt;vec|M|, |N|&gt;)
-  <tr><td>mat2x2&lt;f32&gt;
+  <tr><td>[=matrix|mat=]2x2&lt;f32&gt;
```
[the remark this answers is not in the document]
Comment (dj2) [reply wgsl/index.bs]: done (although it looks kinda weird?)

**Review by alan-baker — APPROVED** [review]

**Comment by dj2** [comment]
@dneto0 ping

File: wgsl/index.bs [hunk 5]
```diff
@@ -2467,7 +2467,7 @@ to WGSL.
     struct small_stride {
       a: array<f32,8> // stride 4
     }
-    @group(0) @binding(0) var<uniform> invalid: small_stride; // Invalid
+    @group(0) @binding(0) var<uniform> invalid: small_stride; // Invalid, stride must be a multiple of 16
```
Comment (dneto0) [inline wgsl/index.bs:2470]: This probably wraps around badly. Please put the comment before the var declaration.

File: wgsl/index.bs [hunk 6]
```diff
@@ -1401,7 +1401,7 @@ Example:  `let minint = -2147483648;` is analyzed as follows:
 
 ## Plain Types ## {#plain-types-section}
 
-[=Plain types=] are the types the machine representation of boolean values, numbers, vectors,
+[=Plain types=] are the machine representation of boolean values, numbers, vectors,
```
Comment (dneto0) [inline wgsl/index.bs:1404]: Should be

> [=Plain types=] are types for the machine representation of ....

**Review by dneto0 — CHANGES_REQUESTED** [review]

File: wgsl/index.bs (hunk 6 above)
Comment (dj2) [reply wgsl/index.bs]: done

File: wgsl/index.bs [hunk 7]
```diff
@@ -2525,9 +2525,9 @@ In code examples in this specification, the comments show this fully parameteriz
 However, in WGSL *source* text:
 * Reference types must not appear.
 * Pointer types may appear.  A pointer type is spelled with parameterization by:
-    * address space,
-    * store type, and
-    * sometimes by access mode, as specified in [[#access-mode-defaults]].
+    * [=address space=],
+    * [=storable|store type=], and
```
[the remark this answers is not in the document]
Comment (dj2) [reply wgsl/index.bs]: done

File: wgsl/index.bs (hunk 5 above)
Comment (dj2) [reply wgsl/index.bs]: done

File: wgsl/index.bs (hunk 6 above)
Comment (dj2) [reply wgsl/index.bs]: done

File: wgsl/index.bs (hunk 7 above)
[the remark this answers is not in the document]
Comment (dj2) [reply wgsl/index.bs]: done

File: wgsl/index.bs [hunk 8]
```diff
@@ -2159,7 +2162,10 @@ following table:
   <tr><td>vec4&lt;|T|&gt;, |T| is [=i32=], [=u32=], or [=f32=]
       <td>16
       <td>16
-  <tr><td>[=matrix|mat=]|N|x|M| (col-major)<br>
+  <tr><td>vec4&lt;f16&gt;
+      <td>8
+      <td>8
+  <tr><td>mat|N|x|M| (col-major)<br>
```
Comment (dneto0) [inline wgsl/index.bs:2168]: These will be |C|x|R|.  That's just merged.

**Review by dneto0 — CHANGES_REQUESTED** [review]

**Comment by dneto0** [comment]
Fails to build

**Comment by dj2** [comment]
Github editing and bot based testing is always fun. Will sync in the creation time expressions PR.

**Review by dneto0 — APPROVED** [review]
Thanks!

**Comment by kdashg** [comment]
<details><summary><a href="https://github.com/gpuweb/gpuweb/wiki/wgsl-2022-04-12-minutes#wgsl-minor-formatting-and-language-changes-2726">WGSL meeting minutes 2022-04-12</a></summary>



* DN: This is editorial.
* DS: Had one fix to apply from David.  Applied it.
* _Merge when conflicts done._


</details>

**Pull request merged**